src/main/java/net/minecraftforge/items/ItemStackHandler.java

```diff
@@ -0,0 +1,155 @@
+package net.minecraftforge.items;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraftforge.common.util.Constants;
+import net.minecraftforge.common.util.INBTSerializable;
+
+public class ItemStackHandler implements IItemHandler, IItemHandlerModifiable, INBTSerializable<NBTTagCompound> {
+    int size;
```

Comment:
A note on this whole class: remember your {s on new lines and it may be worth a 'checkIndex(index){if (index > max) throw DescriptiveError' instead of a index out of bounds.